src/navigators/LoggedInStack.tsx

```diff
@@ -79,9 +93,41 @@ export function LoggedInStack() {
     loadingConsents,
     loadingJoinCircles,
     loadingOnboardingCourse,
+    shouldWaitForOnUserSignIn,
   ]);
 
   const loadingMessage = getLoadingMessage();
+
+  useEffect(() => {
+    const executeOnUserSignInIfNeeded = async () => {
+      if (
+        shouldWaitForOnUserSignIn &&
+        hasAccountAndProject &&
+        hasUserAndPatient
+      ) {
+        await onUserSignIn?.({
+          userId: userData.id,
+          account: account.id,
+          email: userData.profile.email,
+          patientId: activeSubjectId,
+          project: activeProject.id,
```

Comment:
I dig this as an initial, hopefully final, list of data to pass into this func.  Just to say it out loud, the implementor of `onUserSignIn` can also use hooks like `useUser` - so I think it'll be wise of us to not add more data (esp. involving network requests) to these props.  With this in mind, we could actually remove the `userData` additions in this file/PR if we wanted to be as lean as possible.